blocksuite/affine/blocks/code/src/preview/renderer/html.ts

```diff
@@ -0,0 +1,57 @@
+import { html } from 'lit';
+
+import { CodeBlockPreviewExtension } from '../code-preview-extension';
+
+export const CodeBlockHtmlPreview = CodeBlockPreviewExtension('html', model => {
+  if (model.props.language !== 'html') return null;
+
+  const abortController = new AbortController();
+  window.addEventListener(
+    'message',
+    (evt: MessageEvent) => {
+      if (evt.data?.type === 'code-block-html-iframe-height') {
+        document
+          .querySelectorAll(`[data-block-id="${model.id}"] iframe`)
+          .forEach(iframe => {
+            if ((iframe as HTMLIFrameElement).contentWindow === evt.source) {
+              iframe.setAttribute(
+                'style',
+                `width:100%; border:none; height:${evt.data.height}px`
+              );
+            }
+          });
+      }
+    },
+    { signal: abortController.signal }
+  );
+  model.deleted.subscribe(() => {
+    abortController.abort();
+  });
```

Comment:
This codes runs within the scope of the lit render function in `code-block.ts`, and there are some external changes that trigger re-rendering, causing the listeners in this section to be registered repeatedly, such as a new paragraph or other block is added.